src/ol/source/rastersource.js

```diff
@@ -219,32 +219,33 @@ ol.source.Raster.prototype.getImage = function(extent, resolution, pixelRatio, p
     return null;
   }
 
-  if (!this.isDirty_(extent, resolution)) {
+  var currentExtent = ol.object.assign({}, extent);
```

Comment:
How about `var currentExtent = extent.slice()` here instead?  This way we end up with the type we expect.